### PR TITLE
97 implement basic post UI component for the posts feed

### PR DESCRIPTION
**Pull request by MattVwaves**

### CODE REVIEW

File: src/components/posts/PostsPage.js [hunk 1]
```diff
@@ -3,14 +3,17 @@ import { useNavigate } from 'react-router-dom'
...
-import Button from '@mui/material/Button'
 
 import Header from '../Header/Header'
+import Post from './Post'
 
 const PostsPage = ({ loggedInUser }) => {
-  const [post, setPost] = useState({ content: '' })
+  const initialPostState = {
+    content: '',
+    firstName: '',
```
Comment (dogezen) [inline src/components/posts/PostsPage.js:12]: do you need firstName/lastName in post state? you can get this info from loggedInUser

File: src/components/posts/PostsPage.js [hunk 2]
```diff
@@ -19,23 +22,27 @@ const PostsPage = ({ loggedInUser }) => {
...
   const createPost = async (event) => {
     event.preventDefault()
+
     if (post.content.length > 0) {
       client
         .post('/posts', post)
         .then((res) => {
-          console.log(res)
-          setPostResponse(res.data)
+          console.log('created post', res)
```
Comment (dogezen) [inline src/components/posts/PostsPage.js:36]: remove console log

File: src/components/posts/PostsPage.js [hunk 3]
```diff
@@ -44,7 +51,6 @@ const PostsPage = ({ loggedInUser }) => {
   }
   const addPostToFeed = (post) => {
     posts.unshift(post)
```
Comment (dogezen) [inline src/components/posts/PostsPage.js:51]: this won't update state... you need to call `setPosts([post, ...posts])` or similar; make sure you have tested this properly...

File: src/components/posts/PostsPage.js [hunk 4]
```diff
@@ -74,77 +78,24 @@ const PostsPage = ({ loggedInUser }) => {
             <button id="user-signout-button" onClick={signOut}>
               sign out
             </button>
-            <span>Status: {postResponse.status}</span>
+            <span>Status: {postResponse} </span>
```
Comment (dogezen) [inline src/components/posts/PostsPage.js]: you might want to conditionally render the status: ie' if the post has never been submitted, this span can disappear?

File: src/components/posts/style.css [hunk 5]
```diff
@@ -21,7 +21,7 @@
   gap: 20px;
 }
 
-.profile-picture-container,
+/* .profile-picture-container,
```
Comment (dogezen) [inline src/components/posts/style.css:24]: remove commented out css?

File: src/components/posts/PostsPage.js [hunk 6]
```diff
@@ -62,77 +68,27 @@ const PostsPage = ({ loggedInUser }) => {
         <div className="left-sidebar"></div>
         <div className="posts-container">
           <section className="posts-section">
+            <button id="user-signout-button" onClick={signOut}>
```
Comment (dogezen) [inline src/components/posts/PostsPage.js]: please remove this signout button; it was deleted on main branch a few commits ago

File: src/components/cohortsList/CohortsList.js [hunk 7]
```diff
@@ -17,7 +17,18 @@ function CohortsList({ renderHeader, renderAddButton }) {
   }, [])
 
   const header = <h2>Cohorts</h2>
-
+  // const temporaryAddPopup = (
```
[the remark this answers is not in the document]
Comment (MattVwaves) [reply src/components/cohortsList/CohortsList.js:20]: I'm not sure, perhaps I shouldn't have commented it out: this was written by another team but I got a warning that the function was declared but not used. 
Is it ok to have warnings?

File: src/components/posts/PostsPage.js [hunk 8]
```diff
@@ -3,14 +3,17 @@ import { useNavigate } from 'react-router-dom'
...
-import CommentsList from './CommentsList'
-import TextField from '@mui/material/TextField'
-import Button from '@mui/material/Button'
 
 import Header from '../Header/Header'
+import Post from './Post'
 
 const PostsPage = ({ loggedInUser }) => {
-  const [post, setPost] = useState({ content: '' })
+  const initialPostState = {
```
[the remark this answers is not in the document]
Comment (MattVwaves) [reply src/components/posts/PostsPage.js:10]: done this

File: src/components/posts/PostsPage.js [hunk 9]
```diff
@@ -19,23 +22,27 @@ const PostsPage = ({ loggedInUser }) => {
   let navigate = useNavigate()
 
   useEffect(() => {
-    client.get('/posts').then((res) => setPosts(res.data.data.posts))
+    client.get('/posts').then((res) => {
+      console.log(res.data.data.posts)
```
[the remark this answers is not in the document]
Comment (MattVwaves) [reply src/components/posts/PostsPage.js:24]: done

File: src/components/posts/PostsPage.js (hunk 1 above)
Comment (MattVwaves) [reply src/components/posts/PostsPage.js:12]: nope: removed

File: src/components/posts/PostsPage.js (hunk 2 above)
Comment (MattVwaves) [reply src/components/posts/PostsPage.js:36]: done

**Review by dogezen — APPROVED** [review]

**Pull request merged**